pkg/util/push/push.go

```diff
@@ -84,6 +94,33 @@ func handler(maxRecvMsgSize int,
 				logger = log.WithSourceIPs(source, logger)
 			}
 		}
+
+		// TODO: move this to middleware or better place
+		cluster, replica := r.Header.Get(ClusterHeader), r.Header.Get(ReplicaHeader)
+		userID, err := tenant.TenantID(ctx)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
```

Comment:
As mentioned above, let's avoid doing this: we want to shortcut the request, not make important decisions. If we can't decide that we can shortcut, ignore.